execute/plugin.go

```diff
@@ -247,17 +247,20 @@ func (p *Plugin) Observation(
 		return exectypes.NewObservation(groupedCommits, messages, nil).Encode()
 
 	case exectypes.Filter:
-		// TODO: add in nonces, other data comes from previous outcome.
+		// Phase 3: observe nonce for each unique source/sender pair.
 		nonceRequestArgs := make(map[cciptypes.ChainSelector]map[string]struct{})
 
 		// Collect unique senders.
-		for _, commitReport := range previousOutcome.Report.ChainReports {
-			if _, ok := nonceRequestArgs[commitReport.SourceChainSelector]; !ok {
-				nonceRequestArgs[commitReport.SourceChainSelector] = make(map[string]struct{})
+		for _, commitReport := range previousOutcome.PendingCommitReports {
```

Comment:
The report is no longer available at this point, need to grab nonces for the pending commit reports. This may grab more nonces than strictly required.